depends/packages/boost.mk

```diff
@@ -22,7 +22,7 @@ $(package)_config_opts_armv7a_android=address-model=32
 $(package)_toolset_$(host_os)=gcc
 $(package)_archiver_$(host_os)=$($(package)_ar)
 $(package)_toolset_darwin=clang-darwin
-$(package)_config_libraries=chrono,filesystem,system,thread,test
+$(package)_config_libraries=filesystem,system,thread,test
```

Comment:
We also/still depend on boost datetime, but it seems this one is a dependency of boost thread, so implicitly included here?


See also the `apt` command that installs datetime as a dependency of thread:


```
# apt install libboost-thread-dev
Reading package lists... Done
Building dependency tree       
Reading state information... Done
The following additional packages will be installed:
  libboost-atomic1.71-dev libboost-atomic1.71.0 libboost-chrono1.71-dev libboost-chrono1.71.0
  libboost-date-time1.71-dev libboost-date-time1.71.0 libboost-serialization1.71-dev libboost-serialization1.71.0
  libboost-thread1.71-dev libboost-thread1.71.0

Reply:
Perhaps, but I think we should only declare the libraries we explicitly need (just in case the dependency edges get pruned in the future).

Also, according to [the docs for our version of Boost.Thread](https://www.boost.org/doc/libs/1_70_0/doc/html/thread/build.html)

>  Boost.DateTime: This dependency is mandatory, but even if Boost.DateTime is a non header-only library Boost.Thread uses only parts that are header-only, so in principle you should not need to link with the library.